### PR TITLE
add VocabData data_type

### PR DESCRIPTION
**Pull request by ajtritt**
This adds a column type that can be used for storing controlled-vocabulary data.

### CODE REVIEW

File: common/table.yaml [hunk 1]
```diff
@@ -99,6 +99,17 @@ datasets:
...
+- data_type_def: VocabData
+  data_type_inc: VectorData
+  dtype: uint8
+  doc: Data that come from a controlled vocabulary
+  attributes:
+  - name: vocabulary
+    dtype: text
+    shape:
+      - null
+    doc: the available items in the controlled vocabulary
```
Comment (rly) [inline common/table.yaml:111]:
```suggestion
    doc: The available items in the controlled vocabulary. 
```

File: common/table.yaml [hunk 2]
```diff
@@ -99,6 +99,17 @@ datasets:
     dtype: text
     doc: Description of what this table region points to.
 
+- data_type_def: VocabData
+  data_type_inc: VectorData
+  dtype: uint8
+  doc: Data that come from a controlled vocabulary
```
Comment (rly) [inline common/table.yaml:105]:
```suggestion
  doc: Data that come from a controlled vocabulary. A data value of i corresponds
    to the i-th element in the 'vocabulary' array attribute.
```

**Comment by bendichter** [comment]
Though remind me, what's the advantage of doing it this way instead of just using HDF5's built-in enumerated types?

**Comment by ajtritt** [comment]
@bendichter 

> Though remind me, what's the advantage of doing it this way instead of just using HDF5's built-in enumerated types?

After the complications between HDF5 references and Zarr, I'm hesitant to add an enum dtype to HDMF. It would also require more tooling to handle a new enum dtype. This is just simpler and easier to implement.

File: common/namespace.yaml [hunk 3]
```diff
@@ -18,4 +18,4 @@ namespaces:
     title: Table data types
   - doc: data types for different types of sparse matrices
     source: sparse.yaml
-  version: 1.1.3
+  version: 1.2.0
```
Comment (rly) [inline common/namespace.yaml:21]:
```suggestion
  version: 1.2.0a
```

The "master" branch should have a version that differentiates it from public releases. I will write up a release process guide in a separate PR.

Comment (rly) [reply common/namespace.yaml:21]: Release notes should also be updated.

**Comment by rly** [comment]
This PR is good to go. I've added release notes for this in #24.

**Comment by rly** [comment]
I recommend getting this tested and working in HDMF before merging here though.

**Pull request merged**